### PR TITLE
Pass sourcemapConfig to the underlying engine

### PR DESCRIPTION
**Pull request by chancancode**
This allows setting additional options like `sourceRoot`.

### CODE REVIEW

File: concat.js [hunk 1]
```diff
@@ -38,6 +40,8 @@ function ConcatWithMaps(inputNode, options, Strategy) {
   this.footerFiles = options.footerFiles;
   this.separator = (options.separator != null) ? options.separator : '\n';
 
+  delete this.sourceMapConfig.enabled;
```
Comment (rwjblue) [inline concat.js]: won't this mutate the passed in `sourceMapConfig`?

Comment (chancancode) [inline concat.js]: Ah, I see. I can clone it here.

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "broccoli-kitchen-sink-helpers": "^0.2.5",
     "fast-sourcemap-concat": "0.2.7",
     "lodash.merge": "^3.3.2",
+    "lodash.omit": "^3.1.0",
```
Comment (chancancode) [inline package.json]: Why does lodash.omit not have a version 3.3.2? I have no idea.

File: concat.js [hunk 3]
```diff
@@ -71,12 +74,12 @@ ConcatWithMaps.prototype.build = function() {
 
   mkdirp.sync(path.dirname(outputFile));
 
-  this.concat = new this.Strategy({
+  this.concat = new this.Strategy(merge(this.sourceMapConfig, {
```
Comment (rwjblue) [inline concat.js]: Doesn't this also mutate?

**Comment by stefanpenner** [comment]
```
➜  broccoli-concat git:(master) git tag v2.1.0
➜  broccoli-concat git:(master) git push origin v2.1.0
Total 0 (delta 0), reused 0 (delta 0)
nTo git@github.com:ember-cli/broccoli-concat.git
 * [new tag]         v2.1.0 -> v2.1.0
➜  broccoli-concat git:(master) npm publish
+ broccoli-concat@2.1.0
```

**Pull request merged**